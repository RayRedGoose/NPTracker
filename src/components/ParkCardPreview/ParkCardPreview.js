import './ParkCardPreview.scss'
import React from 'react'
import starEmpty from 'assets/star-empty.svg'
import arrow from 'assets/more.svg'

const ParkCardPreview = ({id, image, name}) => {
  return (
    <section id={id} className="park-card">
      <div className="preview" style={{backgroundImage: `url(${image})`}}>
        <header>
          <img src={starEmpty} alt="empty star"/>
          <img src={arrow} alt="arrow"/>
        </header>
      </div>
      <h2>{ name }</h2>
    </section>
  )
}

export default ParkCardPreview
