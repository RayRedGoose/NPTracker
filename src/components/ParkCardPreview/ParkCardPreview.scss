@import "../../variables.scss";
@import "../../index.scss";

.park-card {
  background-color: $darken-small;
  padding: 20px;
  width: 30%;
  margin-bottom: 20px;
  .preview {
    width: 100%; height: 200px;
    background-position: center;
    background-size: cover;
    background-repeat: no-repeat;
    header {
      background-color: $darken-medium;
      display: flex;
      justify-content: space-between;
      padding: 10px;
      img {
        cursor: pointer;
        width: 20px;
        @extend %click-action;
        margin: 0 10px;
      }
    }
  }
  h2 {
    color: $accent-color-primary-light;
    font-family: "Salsa";
    letter-spacing: 0.1em;
    margin-top: 20px;
    text-align: center;
  }
}
