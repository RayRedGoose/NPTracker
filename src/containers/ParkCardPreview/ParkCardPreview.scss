@import "../../variables.scss";
@import "../../index.scss";

.park-card {
  background-color: $darken-small;
  padding: 20px;
  width: 30%; height: fit-content;
  margin: 0 17px 20px;
  .preview {
    width: 100%; height: 200px;
    background-position: center;
    background-size: cover;
    background-repeat: no-repeat;
    header {
      background-color: $darken-medium;
      display: flex;
      justify-content: space-between;
      padding: 10px;
      section {
        display: flex;
        align-items: center;
      }
      .location {
        display: flex;
        align-items: center;
        @extend %location;
        color: $light-color;
        margin-top: 0;
        margin-right: 20px;
        font-size: 20px;
      }
      img {
        cursor: pointer;
        width: 20px;
        @extend %click-action;
        margin-right: 10px;
      }
    }
  }
  .title {
    color: $accent-color-primary-light;
    display: flex;
    justify-content: center;
    font-family: "Salsa";
    font-size: 24px;
    letter-spacing: 0.1em;
    text-align: center;
    text-decoration: none;
    margin-top: 20px;
    height: 90px;
    &:hover {
      text-shadow: 0 0 10px $accent-color-secondary;
    }
  }
}
