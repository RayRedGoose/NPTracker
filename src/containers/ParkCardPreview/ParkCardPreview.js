import './ParkCardPreview.scss'
import React, { Component } from 'react'
import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'
import { NavLink } from 'react-router-dom'
import { addPlannedPark, removePlannedPark, removeProcess } from 'redux/actions'
import { addItem, addItemToAll } from '_utils/localStorage'
// import done from 'assets/done.svg'
import bell from 'assets/bell.svg'
import bellFull from 'assets/bell-full.svg'

export class ParkCardPreview extends Component {
  toggleWishLish = () => {
    const { plannedParks, name } = this.props
    return (!plannedParks.includes(name))
      ? this.addToWishList(name)
      : this.removeFromWishList(name, plannedParks)
  }

  addToWishList = (name) => {
    this.props.addPlannedPark(name)
    addItemToAll('planning', name)
  }

  removeFromWishList = (name, plannedParks) => {
    this.props.removePlannedPark(name)
    const newPlans = plannedParks.filter(park => park !== name)
    addItem('planning', newPlans)
  }

  redirect = () => {
    this.props.removeProcess()
  }

  render() {
    const { id, image, fullName, name, states, plannedParks } = this.props
    const title = (fullName.length > 35 ) ? name : fullName
    const link = fullName.toLowerCase().replace(/ /g, '-')
    const pathForLink = `${this.props.type}/${link}`
    const bellType = (plannedParks.includes(name)) ? bellFull : bell
    return (
      <section id={id} className="park-card">
        <div className="preview" style={{backgroundImage: `url(${image})`}}>
          <header>
            <p className="location">{states}</p>
            <section>
              {
                // <img src={done} alt="flag"/>
              }
              <img src={bellType} alt="bell" onClick={this.toggleWishLish}/>
            </section>
          </header>
        </div>
        <NavLink
          to={pathForLink}
          onClick={this.redirect}
          className="title">{ title }</NavLink>
      </section>
    )
  }
}

export const mapStateToProps = ({plannedParks}) => ({
  plannedParks
})

export const mapDispatchToProps = dispatch => (
  bindActionCreators({
    addPlannedPark,
    removePlannedPark,
    removeProcess
  }, dispatch)
)

export default connect(mapStateToProps, mapDispatchToProps)(ParkCardPreview)
