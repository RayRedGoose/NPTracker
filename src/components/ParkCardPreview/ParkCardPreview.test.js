import React from 'react'
import { shallow } from 'enzyme'
import ParkCardPreview from './ParkCardPreview'

it("should match snapshot with all data passed in correctly", () => {
  const card = shallow(
    <ParkCardPreview
      id={1}
      image="https://www.some.url/preview.jpg"
      name="Some National Park"/>
  )
  expect(card).toMatchSnapshot()
})
