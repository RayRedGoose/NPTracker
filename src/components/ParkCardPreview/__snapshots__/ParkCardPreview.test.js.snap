// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot with all data passed in correctly 1`] = `
<section
  className="park-card"
  id={1}
>
  <div
    className="preview"
    style={
      Object {
        "backgroundImage": "url(https://www.some.url/preview.jpg)",
      }
    }
  >
    <header>
      <section>
        <img
          alt="flag"
          src="done.svg"
        />
        <img
          alt="bell "
          src="bell.svg"
        />
      </section>
      <img
        alt="arrow"
        src="more.svg"
      />
    </header>
  </div>
  <h2>
    Some National Park
  </h2>
</section>
`;
